### ISSUE TITLE
Query parameter serialization regression from 2.18.0 to 3.0.0-beta.1

### ISSUE THREAD

**Issue by mwpastore** [emberjs/ember.js#16053]
This works fine in Ember 2.18.0:

```javascript
import Route from '@ember/routing/route';

import { inject as service } from '@ember/service';

export default Route.extend({
  ajax: service(),

  model() {
    return this.get('ajax').request('/foo', {
      data: {
        bar: [
          'qux'
        ]
      }
    });
  }
});
```

The bar query parameter is serialized as an array, and the generated, encoded URL looks like `http://localhost:4200/foo?bar%5B%5D=qux`.

In 3.0.0-beta.1, the same model hook raises a TypeError with the following stack trace:

```
TypeError: Cannot convert a Symbol value to a string
    at messageFor (http://localhost:4200/assets/vendor.js:33771:62)
    at Object.get (http://localhost:4200/assets/vendor.js:33789:49)
    at Proxy.toString (<anonymous>)
    at Function.isPlainObject (http://localhost:4200/assets/vendor.js:644:25)
    at Function.jQuery.extend.jQuery.fn.extend (http://localhost:4200/assets/vendor.js:577:35)
    at Function.jQuery.extend.jQuery.fn.extend (http://localhost:4200/assets/vendor.js:589:30)
    at Function.jQuery.extend.jQuery.fn.extend (http://localhost:4200/assets/vendor.js:589:30)
    at ajaxExtend (http://localhost:4200/assets/vendor.js:8945:10)
    at Function.ajaxSetup (http://localhost:4200/assets/vendor.js:9193:4)
    at ajax (http://localhost:4200/assets/vendor.js:9242:15)
```

I am taking a freshly-generated Ember CLI 3.0.0-beta.1 repo and just toggling the ember-source dependency to get a working/non-working version. The jQuery and ember-ajax versions are the same for both. Other types of query parameter values, such as strings, numbers, and non-array objects, seem to be serializing correctly.

**Comment by rwjblue** [comment]
@mwpastore - Could you publish that demo repo so I can test out a possible fix?

**Comment by rwjblue** [comment]
FWIW, this is related to https://github.com/emberjs/ember.js/issues/16049

**Comment by rwjblue** [comment]
A recent commit I pushed to https://github.com/emberjs/ember.js/pull/16050 will fix the `TypeError`, but without an actual repro I can't confirm/deny that the whole thing works...

**Comment by mwpastore** [comment]
@rwjblue That was quick! 

It's quite literally an `ember new` repo with that one route. I've pushed it [here](https://github.com/mwpastore/ember-ajax-symbol).

**Comment by mwpastore** [comment]
@rwjblue I tried your PR with my repro repo (say that ten times fast) and now I'm getting:

```
TypeError: Cannot set property firstObject of [object Array] which has only a getter
    at Function.jQuery.extend.jQuery.fn.extend (jquery.js:254)
    at Function.jQuery.extend.jQuery.fn.extend (jquery.js:250)
    at Function.jQuery.extend.jQuery.fn.extend (jquery.js:250)
    at ajaxExtend (jquery.js:8606)
    at Function.ajaxSetup (jquery.js:8854)
    at ajax (jquery.js:8903)
    at Class._makeRequest (ajax-request.js:252)
    at Class.request (ajax-request.js:196)
    at Class.model (application.js:11)
    at Class.deserialize (route.js:942)
```

So... progress?

**Comment by rwjblue** [comment]
After digging, this is essentially the same issue as the following:

* #16141
* #16053
* #16049

**Comment by rwjblue** [comment]
I believe this is fixed in 3.0.0-beta.3. Closing for now, but happy to reopen if I'm mistaken...
